index.bs

```diff
@@ -402,9 +565,29 @@ perform the following steps:
 
 </div>
 
-Every [=bounce tracking timer period=] the user agent should run the
-[=bounce tracking timer=] algorithm given the [=wall clock=]'s
-[=wall clock/unsafe current time=].
+This algorithm is called from [=process response received for bounce tracking=],
+after a timeout of [=client bounce detection timer period=], unless it is cancelled
+by toggling [=bounce tracking record/client bounce detection running=].
+
+<div algorithm>
+
+To run the <dfn>record stateful bounces for bounce tracking</dfn> algorithm
+given a [=navigable=] |navigable|, perform the following steps:
+
+1. If |navigable|'s [=top-level traversable/bounce tracking record=]'s [=bounce tracking record/client bounce detection running=] is false, abort these steps.
+1. [=map/For each=] |host| in |navigable|'s [=top-level traversable/bounce tracking record=]'s [=bounce tracking record/bounce set=]:
```

Comment:
This assumes |navigable| is a top-level traversable, while the "Let |topDocument|" step assumes it might not be. Which is it?